src/hotspot/cpu/ppc/macroAssembler_ppc.cpp

```diff
@@ -1438,12 +1438,16 @@ bool MacroAssembler::is_load_from_polling_page(int instruction, void* ucontext,
     return true; // No ucontext given. Can't check value of ra. Assume true.
   }
 
-#ifdef LINUX
+#if defined(LINUX) || defined(_ALLBSD_SOURCE)
   // Ucontext given. Check that register ra contains the address of
   // the safepoing polling page.
   ucontext_t* uc = (ucontext_t*) ucontext;
   // Set polling address.
+#if defined(LINUX)
   address addr = (address)uc->uc_mcontext.regs->gpr[ra] + (ssize_t)ds;
+#elif defined(_ALLBSD_SOURCE)
+  address addr = (address)uc->uc_mcontext.mc_gpr[ra] + (ssize_t)ds;
+#endif
```

Comment:
As far as I can tell, this code dealing wih `ucontext` is unreachable.

Its only caller is
```
  bool is_safepoint_poll() {
    // The current arguments of the instruction are not checked!
    if (USE_POLL_BIT_ONLY) {
      int encoding = SafepointMechanism::poll_bit();
      return MacroAssembler::is_tdi(long_at(0), Assembler::traptoGreaterThanUnsigned | Assembler::traptoEqual,
                                    -1, encoding);
    }
    return MacroAssembler::is_load_from_polling_page(long_at(0), nullptr);
  }

```